packages/electron-updater/src/AppUpdater.ts

```diff
@@ -248,11 +248,19 @@ function createClient(data: string | PublishConfiguration | BintrayOptions | Git
   switch (provider) {
     case "github":
       return new GitHubProvider(<GithubOptions>data)
+    case "s3":
     case "generic":
+      if (provider === "s3") {
+        let s3Data = {
```

Comment:
I don't see any error — tslint or IDEA are happy.

But in any case consider to do it in a separate branch: `new GenericProvider` is not a duplicated code :)